lib/pbio/src/light/animation.c

```diff
@@ -117,10 +117,15 @@ PROCESS_THREAD(pbio_light_animation_process, ev, data) {
         PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_TIMER);
         struct etimer *timer = data;
         pbio_light_animation_t *animation = PBIO_CONTAINER_OF(timer, pbio_light_animation_t, timer);
-        clock_time_t interval = animation->next(animation);
-        if (pbio_light_animation_is_started(animation)) {
-            etimer_reset_with_new_interval(&animation->timer, interval);
+
+        // Like all etimer events, only handle expired etimers. Need this check
+        // in case the timer was stopped or reset before this event was handled.
+        if (!etimer_expired(&animation->timer) || !pbio_light_animation_is_started(animation)) {
```

Comment:
It seems like continuing when timer is expired (due to animation not started) could be problematic since we wouldn't get another timer event. Should we be resetting the timer in that case?

Reply:
Isn't this the same as how it was, wrapping around the loop if there was nothing to do?